ibis/expr/rules.py

```diff
@@ -90,22 +97,17 @@ def just(arg):
     return lambda **_: arg
 
 
+# TODO(kszucs): perhaps rename to nodes_of
 @rule
-def value_list_of(inner, arg, **kwargs):
-    # TODO(kszucs): would be nice to remove ops.ValueList
+def list_of(inner, arg, **kwargs):
```

Comment:
This rule must be used to define a sequence of nodes which automatically traversed by other utilities. We already have a `tuple_of` rule representing a python tuple of arbitrary types so this naming might be a bit misleading, the reason I chose this was that to align with the operation name `ops.List`.

@cpcloud opinions?

Reply:
Can we rename `ops.List` to `ops.NodeList` and `list_of` to `node_list_of`?

Reply:
Can you rename? I'll merge after that and the comment below on removing the assertion.

Reply:
Sure